utils/rust-setup.js

```diff
@@ -0,0 +1,122 @@
+const chalk = require('chalk');
+const os = require('os');
+const readline = require('readline');
+const sh = require('shelljs');
+
+// script from https://rustup.rs/ with auto-accept flag "-y"
+const installRustupScript = "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y";
+const updatePath = 'source $HOME/.cargo/env';
+const addWasm32TargetScript = "rustup target add wasm32-unknown-unknown";
+// We should update PATH in the same script because every new Bash scripts are executed in a separate shell
+const updatePathAndAddWasm32TargetScript = updatePath + ' && ' + addWasm32TargetScript;
+
+const isWindows = os.platform() == 'win32'
+
+function isRustupInstalled() {
+    console.log(chalk`Checking if {bold rustup} is installed...`);
+    const isInstalled = sh.exec('rustup --version &> /dev/null').code == 0;
+    console.log(chalk`{bold rustup} is`, isInstalled ? 'installed' : 'not installed');
+    return isInstalled;
+}
+
+function isWasmTargetAdded() {
+    const addedTargets = sh.exec('rustup target list --installed').stdout;
+    const isWasmTargetAdded = addedTargets.includes('wasm32-unknown-unknown');
+    console.log(chalk`{bold wasm32-unknown-unknown} target `, isWasmTargetAdded ? 'already added' : 'is not added');
+    return isWasmTargetAdded;
+}
+
+function installRustup() {
+    console.log(chalk`Installing {bold rustup}...`);
+    sh.exec(installRustupScript);
+}
+
+function addWasm32Target() {
+    console.log(chalk`Adding {bold wasm32-unknown-unknown} target...`);
+    sh.exec(updatePathAndAddWasm32TargetScript);
+}
+
+async function askYesNoQuestionAndRunFunction(question, functionToRun = null) {
+    const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+    });
+    try {
+        for (let attempt = 0; attempt < 4; attempt++) {
+            const answer = await new Promise((resolve) => {
+                rl.question(question, async (userInput) => {
```

Comment:
```suggestion
                rl.question(question, (userInput) => {
```

Doesn't look like there are any `await`s in this function